src/main/java/org/infinispan/tutorial/db/LocationWeatherMarshallingContext.java

```diff
@@ -2,44 +2,31 @@
 
 import org.infinispan.client.hotrod.RemoteCache;
 import org.infinispan.client.hotrod.RemoteCacheManager;
-import org.infinispan.client.hotrod.marshall.MarshallerUtil;
-import org.infinispan.protostream.SerializationContext;
-import org.infinispan.protostream.annotations.ProtoSchemaBuilder;
-import org.infinispan.tutorial.data.LocationWeather;
+import org.infinispan.client.hotrod.impl.query.RemoteQuery;
+import org.infinispan.query.remote.client.ProtobufMetadataManagerConstants;
 
-import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.Objects;
 
-import static org.infinispan.query.remote.client.ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME;
-
 public final class LocationWeatherMarshallingContext {
 
     public static void initSerializationContext(RemoteCacheManager cacheManager) {
         System.out.println("---- LocationWeatherMarshallingContext - initialize the serialization context for LocationWeather class ----");
         Objects.requireNonNull(cacheManager);
 
-        // Get the serialization context of the client
-        SerializationContext ctx = MarshallerUtil.getSerializationContext(cacheManager);
-
-        // Use ProtoSchemaBuilder to define a Protobuf schema on the client
-        ProtoSchemaBuilder protoSchemaBuilder = new ProtoSchemaBuilder();
-        String fileName = "weather.proto";
-        String protoFile = null;
         try {
-            protoFile = protoSchemaBuilder
-                    .fileName(fileName)
-                    .addClass(LocationWeather.class)
-                    .packageName("org.infinispan.tutorial.data")
-                    .build(ctx);
-        } catch (IOException e) {
+            // Retrieve the generated Protobuf schema on the client
+            Path proto = Paths.get(RemoteQuery.class.getClassLoader()
```

Comment:
This piece of code can be simplified. I'll push a commit on top of this PR with my suggested changes